htdocs/societe/class/api_thirdparties.class.php

```diff
@@ -1273,11 +1273,13 @@ public function deleteCompanyBankAccount($id, $bankaccount_id)
 
 		$account->fetch($bankaccount_id);
 
-		if (!$account->socid == $id) {
-			throw new RestException(403);
-		}
+		$socid = (int) $account->socid;
 
-		return $account->delete(DolibarrApiAccess::$user);
+		if ($socid == $id) {
+			return $account->delete(DolibarrApiAccess::$user);
+		} else {
```

Comment:
416 is reserved for the use of range header.
For this use case, best cide seems 403 that may means not allowed (not because peymissions are not granted, but because value of data to delete are not enough consistent to allow access).
A message can be added in the through to help user understand

throw new RestException(403, "Not allowed due to bad consistency of input data");

Reply:
ok